src/optimizer/plan_generation.c

```diff
@@ -4766,3 +4877,41 @@ qo_get_orderby_num_upper_bound_node (PARSER_CONTEXT * parser, PT_NODE * orderby_
   /* Any other comparison operator is unusable */
   return NULL;
 }
+
+/*
+ * qo_get_multi_col_range_segs () -
+ *   return:
+ *   env(in): The optimizer environment
+ *   plan(in): Query plan
+ *   qo_index_infop(in):
+ *   multi_col_segs(out): (a,b) in ... a,b's segment number bit
+ *   multi_col_range_segs(out): range segments in multiple column term
+ *
+ * Note: return multiple column term's number
+ *       output are multi column term's segments and range key filter segments and index col segments
+ */
+static int
+qo_get_multi_col_range_segs (QO_ENV * env, QO_PLAN * plan, QO_INDEX_ENTRY * index_entryp,
+			     BITSET * multi_col_segs, BITSET * multi_col_range_segs, BITSET * index_segs)
+{
+  BITSET_ITERATOR iter;
+  QO_TERM *termp = NULL;
+  int multi_term = -1;
+
+  /* find term having multiple columns ex) (col1,col2) in ... */
+  for (multi_term = bitset_iterate (&(plan->plan_un.scan.terms), &iter); multi_term != -1;
+       multi_term = bitset_next_member (&iter))
+    {
+      termp = QO_ENV_TERM (env, multi_term);
+      if (QO_TERM_IS_FLAGED (termp, QO_TERM_MULTI_COLL_PRED))
+	{
+	  bitset_assign (multi_col_segs, &(QO_TERM_SEGS (termp)));
+	  break;
+	}
+    }
+
+  bitset_assign (index_segs, &(index_entryp->index_segs));
```

Comment:
might need to check if index_entryp is null.

Reply:
i fixed it